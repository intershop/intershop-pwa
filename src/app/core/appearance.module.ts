import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
