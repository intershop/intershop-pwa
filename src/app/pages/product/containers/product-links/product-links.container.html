<ng-container *ngIf="links$ | async as links">
  <ish-product-links-list
    [links]="links.upselling"
    [productLinkTitle]="'product.product_links.upselling.title' | translate"
  ></ish-product-links-list>
  <!---<ish-product-links-carousel [links]="links.crossselling"></ish-product-links-carousel>-->
</ng-container>
