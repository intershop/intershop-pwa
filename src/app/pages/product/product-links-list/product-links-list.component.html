<div *ngIf="links.products.length" class="product-list-container">
  <h2>{{ productLinkTitle }}</h2>
  <div class="product-list">
    <div *ngFor="let sku of links.products" class="product-list-item list-view">
      <ish-product-item ishProductContext [sku]="sku" displayType="row"></ish-product-item>
    </div>
  </div>
</div>
