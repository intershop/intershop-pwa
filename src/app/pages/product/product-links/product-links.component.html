<ng-container *ngIf="links$ | async as links">
  <ng-container *ngIf="links.upselling?.products?.length">
    <ish-product-links-list
      [links]="links.upselling"
      [productLinkTitle]="'product.product_links.upselling.title' | translate"
    ></ish-product-links-list>
  </ng-container>
  <ng-container *ngIf="links.spareparts?.products?.length">
    <ish-product-links-list
      [links]="links.spareparts"
      [productLinkTitle]="'product.product_links.spareparts.title' | translate"
    ></ish-product-links-list>
  </ng-container>
  <ng-container *ngIf="links.crossselling?.products?.length">
    <ish-product-links-carousel
      [links]="links.crossselling"
      [productLinkTitle]="'product.product_links.crossselling.title' | translate"
    ></ish-product-links-carousel>
  </ng-container>
  <ng-container *ngIf="links.accessory?.products?.length">
    <ish-product-links-carousel
      [links]="links.accessory"
      [productLinkTitle]="'product.product_links.accessory.title' | translate"
    ></ish-product-links-carousel>
  </ng-container>
  <ng-container *ngIf="links.differentorderunit?.products?.length">
    <ish-product-links-carousel
      [links]="links.differentorderunit"
      [productLinkTitle]="'product.product_links.differentorderunit.title' | translate"
    ></ish-product-links-carousel>
  </ng-container>
  <ng-container *ngIf="links.followup?.products?.length">
    <ish-product-links-carousel
      [links]="links.followup"
      [productLinkTitle]="'product.product_links.followup.title' | translate"
    ></ish-product-links-carousel>
  </ng-container>
  <ng-container *ngIf="links.other?.products?.length">
    <ish-product-links-carousel
      [links]="links.other"
      [productLinkTitle]="'product.product_links.other.title' | translate"
    ></ish-product-links-carousel>
  </ng-container>
  <ng-container *ngIf="links.replacement?.products?.length">
    <ish-product-links-carousel
      [links]="links.replacement"
      [productLinkTitle]="'product.product_links.replacement.title' | translate"
    ></ish-product-links-carousel>
  </ng-container>
</ng-container>
