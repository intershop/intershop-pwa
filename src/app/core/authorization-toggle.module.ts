import { ModuleWithProviders, NgModule } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { map } from 'rxjs/operators';

import { AuthorizationToggleDirective } from './directives/authorization-toggle.directive';
import { AuthorizationToggleService, checkPermission } from './utils/authorization-toggle/authorization-toggle.service';
import { whenTruthy } from './utils/operators';

@NgModule({
  declarations: [AuthorizationToggleDirective],
  exports: [AuthorizationToggleDirective],
})
export class AuthorizationToggleModule {
  private static permissions = new ReplaySubject<string[]>(1);

  static forTesting(...permissions: string[]): ModuleWithProviders<AuthorizationToggleModule> {
    AuthorizationToggleModule.switchTestingPermissions(...permissions);
    return {
      ngModule: AuthorizationToggleModule,
      providers: [
        {
          provide: AuthorizationToggleService,
          useValue: {
            isAuthorizedTo: (permission: string) =>
              AuthorizationToggleModule.permissions.pipe(
                whenTruthy(),
                map(perms => checkPermission(perms, permission))
              ),
          },
        },
      ],
    };
  }

  static switchTestingPermissions(...permissions: string[]) {
    AuthorizationToggleModule.permissions.next(permissions);
  }
}

export { AuthorizationToggleService } from './utils/authorization-toggle/authorization-toggle.service';
export { AuthorizationToggleGuard } from './guards/authorization-toggle.guard';
