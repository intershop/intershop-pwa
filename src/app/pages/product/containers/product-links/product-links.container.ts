import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
import { Store, select } from '@ngrx/store';
import { Observable } from 'rxjs';

import { ProductLinksView } from 'ish-core/models/product-links/product-links.model';
import { LoadProductLinks, getProductLinks } from 'ish-core/store/shopping/products';

@Component({
  selector: 'ish-product-links-container',
  templateUrl: './product-links.container.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProductLinksContainerComponent implements OnChanges {
  @Input() sku: string;

  links$: Observable<ProductLinksView>;

  constructor(private store: Store<{}>) {}

  ngOnChanges() {
    if (this.sku) {
      this.store.dispatch(new LoadProductLinks({ sku: this.sku }));
      this.links$ = this.store.pipe(select(getProductLinks, { sku: this.sku }));
    }
  }
}
