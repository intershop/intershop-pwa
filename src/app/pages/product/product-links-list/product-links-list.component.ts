import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { ProductLinks } from 'ish-core/models/product-links/product-links.model';

/**
 * The Product Link List Component
 *
 * Displays the products which are assigned to a specific type as an list.
 * It uses the {@link ProductItemContainerComponent} for the rendering of products.
 *
 * @example
 * <ish-product-links-list [links]="links.upselling" [productLinkTitle]="'product.product_links.upselling.title' | translate"></ish-product-links-list>
 */
@Component({
  selector: 'ish-product-links-list',
  templateUrl: './product-links-list.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProductLinksListComponent {
  /**
   * list of products which are assigned to the specific product link type
   */
  @Input() links: ProductLinks;
  /**
   * title that should displayed for the specific product link type
   */
  @Input() productLinkTitle: string;
}
