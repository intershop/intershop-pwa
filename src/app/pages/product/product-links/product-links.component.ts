import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { Observable } from 'rxjs';

import { ProductContextFacade } from 'ish-core/facades/product-context.facade';
import { ProductLinksDictionary } from 'ish-core/models/product-links/product-links.model';

/**
 * The Product Link Container Component
 *
 * Collects all types of product links and prepares their rendering.
 * Product links can be displayed as product list (uses {@link ProductLinkListComponent}) or product carousel (uses  {@link ProductLinksCarouselComponent}).
 * For the carousel swiper is used.
 *
 * @example
 * <ish-product-links-list [links]="links.upselling" [productLinkTitle]="'product.product_links.upselling.title' | translate"></ish-product-links-list>
 */
@Component({
  selector: 'ish-product-links',
  templateUrl: './product-links.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProductLinksComponent implements OnInit {
  links$: Observable<ProductLinksDictionary>;

  constructor(private context: ProductContextFacade) {}

  ngOnInit() {
    this.links$ = this.context.select('links');
  }
}
