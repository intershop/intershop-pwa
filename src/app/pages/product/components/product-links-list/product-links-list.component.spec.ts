import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { TranslateModule } from '@ngx-translate/core';
import { MockComponent } from 'ng-mocks';

import { ProductLinkView } from 'ish-core/models/product-links/product-links.model';
import { ProductItemContainerComponent } from '../../../../shared/product/containers/product-item/product-item.container';

import { ProductLinksListComponent } from './product-links-list.component';

describe('Product Links List Component', () => {
  let component: ProductLinksListComponent;
  let fixture: ComponentFixture<ProductLinksListComponent>;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [TranslateModule.forRoot()],
      declarations: [MockComponent(ProductItemContainerComponent), ProductLinksListComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    const productLink = { productSKUs: ['sku'], categoryIds: ['catID'] } as ProductLinkView;

    fixture = TestBed.createComponent(ProductLinksListComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    component.links = productLink;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchInlineSnapshot(`
      <div class="product-list-container">
        <h2></h2>
        <div class="product-list">
          <div class="product-list-item list-view">
            <ish-product-item-container ng-reflect-product-sku="sku"></ish-product-item-container>
          </div>
        </div>
      </div>
    `);
  });
});
