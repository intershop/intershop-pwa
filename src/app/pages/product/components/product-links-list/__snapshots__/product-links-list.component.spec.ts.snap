// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Links List Component should be created 1`] = `
<div class="product-list-container">
  <h2></h2>
  <div class="product-list">
    <div class="product-list-item list-view">
      <ish-product-item-container
        ng-reflect-product-sku="sku"
        ng-reflect-configuration="[object Object]"
      ></ish-product-item-container>
    </div>
  </div>
</div>
`;
